File: leetcode3/황은지/611. Valid Triangle Number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var triangleNumber = function (nums) {
  // 가장 긴 변이 나머지 두변 합보다 작아야됨
  let count = 0;
  nums.sort((a, b) => b - a);

  for (let i = 0; i < nums.length; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      for (let k = j + 1; k < nums.length; k++) {
        if (nums[i] >= nums[j] + nums[k]) break;
        count++;
      }
    }
  }

  return count;
};

File: leetcode3/남효정/611. Valid Triangle Number.py
# 풀이 실패
class Solution:
    def triangleNumber(self, nums: List[int]) -> int:
        nums.sort()
        cnt = 0
        N = len(nums)
        
        for k in range(N-1, 1, -1):
            left = 0
            right = k - 1

            # 투 포인터
            while left < right:
                if nums[left] + nums[right] > nums[k]:
                    cnt += right - left
                    right -= 1
                else:
                    left += 1

        return cnt
